app/script/view_model/content/InputBarViewModel.js

```diff
@@ -419,21 +419,23 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
 
   handleMentionFlow() {
     const textarea = document.querySelector('#conversation-input-bar-text');
-    const value = textarea.value;
-    const {selectionStart, selectionEnd} = textarea;
+    const {selectionStart, selectionEnd, value} = textarea;
+
     const textInSelection = value.substring(selectionStart, selectionEnd);
     const wordBeforeSelection = value.substring(0, selectionStart).replace(/[^]*\s/, '');
     const isSpaceSelected = /\s/.test(textInSelection);
-    const isOverMention =
-      this.findMentionAtPosition(selectionStart - 1, this.currentMentions()) ||
-      this.findMentionAtPosition(selectionEnd, this.currentMentions());
+
+    const isSelectionStartMention = this.findMentionAtPosition(selectionStart - 1, this.currentMentions());
+    const isSelectionEndMention = this.findMentionAtPosition(selectionEnd, this.currentMentions());
+    const isOverMention = isSelectionStartMention || isSelectionEndMention;
     const isOverValidMentionString = /^@\S*$/.test(wordBeforeSelection);
 
     if (!isSpaceSelected && !isOverMention && isOverValidMentionString) {
       const wordAfterSelection = value.substring(selectionEnd).replace(/\s[^]*/, '');
+
       const term = `${wordBeforeSelection.replace(/^@/, '')}${textInSelection}${wordAfterSelection}`;
-      const start = selectionStart - wordBeforeSelection.length;
-      this.editedMention({start, term});
+      const startIndex = selectionStart - wordBeforeSelection.length;
+      this.editedMention({startIndex, term});
```

Comment:
this will break other occurrences of editedMention

Reply:
Took care of that by renaming the other end as well